packages/@uppy/utils/src/dataURItoBlob.ts

```diff
@@ -1,28 +1,30 @@
 const DATA_URL_PATTERN = /^data:([^/]+\/[^,;]+(?:[^,]*?))(;base64)?,([\s\S]*)$/
 
-export default function dataURItoBlob (dataURI, opts, toFile) {
+export default function dataURItoBlob (dataURI: string, opts: { mimeType: string, name: string }, toFile: boolean) {
```

Comment:
In the case of a library, I would consider it a best practice to use explicit return types, not inferred return types. There's a `typescript-eslint` rule to enforce that. WDYT?

Reply:
not sure if dataURItoBlob is a public API though? we don't document it as part of the uppy api

Reply:
I don't see why it would change anything that we're a library, `tsc` will still produce the appropriate return type when we ask it to emit the `.d.ts` file. Or is it to avoid having unintentional semver-minor change slip through?

Reply:
I think you should not distinct between public vs. private in this API but just make a choice between inferred and explicit and stick with it throughout the codebase. Explicit return types throughout will lead to better generated `.d.ts` files in the end.

Reply:
my two cents: simple functions without conditional return types are fine to infer. Otherwise type it. If that's too inconsistent or forgetful we could add the eslint rule yes.

Reply:
> Explicit return types throughout will lead to better generated `.d.ts` files in the end.

Shouldn't we expect TS to generate better return type than us humans? 🤔

Reply:
Maybe we just don’t force anything using rules, just allow people to do what they prefer?

Reply:
> > Explicit return types throughout will lead to better generated `.d.ts` files in the end.
> 
> Shouldn't we expect TS to generate better return type than us humans? 🤔

Why would an inferred type be better than you describing exactly what you want/need a function to return?

Reply:
Which might not be what you want it to return. If you specify the return type you'll get TS errors if you do not return exactly that.